src/game/Player.cpp

```diff
@@ -15841,8 +15841,9 @@ void Player::SaveToDB()
         _SaveStats();
 
     // save pet (hunter pet level and experience and all type pets health/mana).
-    if (Pet* pet = GetPet())
-        pet->SavePetToDB(PET_SAVE_AS_CURRENT);
+    if (getClass() != CLASS_PRIEST)
```

Comment:
It's not, no, this way may also be lighter on the server actually, so nevermind my comment altogether.